nexus/db-queries/src/db/saga_recovery.rs

```diff
@@ -195,17 +198,32 @@ async fn list_unfinished_sagas(
     // Although we could read them all into memory simultaneously, this
     // risks blocking the DB for an unreasonable amount of time. Instead,
     // we paginate to avoid cutting off availability to the DB.
-    let mut last_id = None;
     let mut sagas = vec![];
-    loop {
-        let pagparams = new_page_params(last_id.as_ref());
-        let mut some_sagas =
-            datastore.saga_list_unfinished_by_id(sec_id, &pagparams).await?;
-        if some_sagas.is_empty() {
-            break;
-        }
-        sagas.append(&mut some_sagas);
-        last_id = Some(sagas.last().as_ref().unwrap().id.0 .0);
+    let mut paginator = Paginator::new(BATCH_SIZE);
+    let conn = datastore.pool_connection_unauthorized().await?;
```

Comment:
Are we not able to use `pool_connection_authorized()` here with the `OpContext` we have?

Reply:
Sure, happy to switch to this.